evm_loader/test_cli_emulate.py

```diff
@@ -106,94 +97,138 @@ def setUpClass(cls):
-                                self.contract.ethereum_id.hex(),
-                                trx_data.hex(),
-                                [
-                                    [self.contract.contract_account,
-                                     '0x' + self.contract.ethereum_id.hex(),
-                                     self.contract.contract_code_account],
-                                    [self.caller,
-                                     '0x' + self.ethereum_caller.hex(),
-                                     None]
-                                ])
-        # no changes after emulate
-        assert (self.tokenBalance(token_acc1) == mint_amount - transfer_amount)
-        assert (self.tokenBalance(token_acc2) == transfer_amount)
+        (trx_data, result) \
+            = self.contract.transfer_ext(self.ethereum_caller,
+                                         self.token, self.token_acc1, self.token_acc2, transfer_amount * (10 ** 9),
+                                         self.acc.public_key()._key)
+        src_data = result['result']['meta']['innerInstructions'][-1]['instructions'][-1]['data']
+        self.assertEqual(base58.b58decode(src_data)[0], 6)  # 6 means OnReturn
+        self.assertLess(base58.b58decode(src_data)[1], 0xd0)  # less 0xd0 - success
+
+        self.assertEqual(self.spl_token.balance(self.token_acc1), balance1 + mint_amount - transfer_amount)
+        self.assertEqual(self.spl_token.balance(self.token_acc2), balance2 + transfer_amount)
+
+        emulate_result = emulate_external_call(self.ethereum_caller.hex(),
+                                               self.contract.ethereum_id.hex(),
+                                               trx_data.hex())
+        tmpl = """ {
+            "accounts": [
+                {
+                    "account": "CONTRACT_ACCOUNT",
+                    "address": "CONTRACT_ETHEREUM_ID",
+                    "code_size": null,
+                    "contract": "CONTRACT_CODE_ACCOUNT",
+                    "new": false,
+                    "writable": true
+                },
+                {
+                    "account": "ETHEREUM_CALLER_ACCOUNT",
+                    "address": "ETHEREUM_CALLER_ID",
+                    "code_size": null,
+                    "contract": null,
+                    "new": false,
+                    "writable": true
+                }
+            ],
+            "exit_status": "succeed",
+            "result": ""
+        }"""
+        tmpl = tmpl.replace('CONTRACT_ACCOUNT', self.contract.contract_account)
+        tmpl = tmpl.replace('CONTRACT_ETHEREUM_ID', '0x' + self.contract.ethereum_id.hex())
+        tmpl = tmpl.replace('CONTRACT_CODE_ACCOUNT', self.contract.contract_code_account)
+        tmpl = tmpl.replace('ETHEREUM_CALLER_ACCOUNT', self.caller)
+        tmpl = tmpl.replace('ETHEREUM_CALLER_ID', '0x' + self.ethereum_caller.hex())
+        tmpl_json = json.loads(tmpl)
```

Comment:
Why do you need to prepare `tmpl_json` from a string? You can build `tmpl_json` object in place like this:
```
        tmpl = {
            "accounts": [
                {
                    "account": self.contract.contract_account,
                    "address": '0x' + self.contract.ethereum_id.hex(),
                    "code_size": None,
                    "contract": self.contract.contract_code_account
                    "new": False,
                    "writable": True
                },
                {
                    "account": self.caller,
                    "address": '0x' + self.ethereum_caller.hex(),
                    "code_size": None,
                    "contract": None,
                    "new": False,
                    "writable": True
                }
            ],
            "exit_status": "succeed",
            "result": ""
        }
```

Reply:
it will be resolved in a next task